### PR TITLE
v6.3 release notes: Add HAProxy note

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Add a note about HAProxy

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
  - https://github.com/pingcap/tidb/pull/27768

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by bb7133 — APPROVED** [review]
LGTM

**Comment by bb7133** [comment]
PTAL @TomShawn @qiancai

File: releases/release-6.3.0.md [hunk 1]
```diff
@@ -249,6 +249,7 @@ In v6.3.0-DMR, the key new features and improvements are as follows:
...
 * To deploy TiFlash under the Linux AMD64 architecture, the CPU must support AVX2 instruction sets. Use `cat /proc/cpuinfo | grep avx2` to confirm that there is output. By using such CPU instruction sets, TiFlash's vectorization engine can deliver better performance.
+* The minimum version of HAProxy that works with TiDB is now v1.5. HAProxy versions between v1.5 and v2.1 now require the `post-41` configuration option to be set in the `mysql-check`. It is recommended to use HAProxy version v2.2 or newer.
```
Comment (shichun-0415) [inline releases/release-6.3.0.md:252]:
```suggestion
* The minimum version of HAProxy that works with TiDB is now v1.5. HAProxy versions between v1.5 and v2.1 now require the `post-41` configuration option to be set in `mysql-check`. It is recommended to use HAProxy v2.2 or newer.
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Pull request merged**